### ISSUE TITLE
[QUESTION] How to disable gesture detector

### ISSUE THREAD

**Issue by themadmrj** [jhomlala/betterplayer#347]
First some code to explain my situation.
I use the last version of the lib 0.0.59.
My Widget looks like this:

```

bool showDetails = false;

Stack(children: [
  GestureDetector(
                onTap: onTap,
                onLongPress: onLongPress,
                child: BetterPlayer(controller: _betterPlayerController)),

 Visibility(
              visible: !showDetails,
              child: Container())
])
```

My controller:

```
_betterPlayerController = BetterPlayerController(BetterPlayerConfiguration(
      autoDetectFullscreenDeviceOrientation: true,
      fit: BoxFit.cover,
      autoPlay:  true,
      looping: true,
      controlsConfiguration: BetterPlayerControlsConfiguration(
          enableOverflowMenu: false,
          enableSkips: false),
    )));
```

I call `_betterPlayerController.setControlsEnabled(false);` to disable the controls, since I'm handling it myself. `onTap` will pause the video, set `showDetails` to `true` and display the details on top of the paused video. Calling `onTap` again will reverse the process, hiding the details and play the video again. `onLongPress` triggers displaying the video in fullscreen. Nothing happens when double tapping.
If the video fails to load, tapping the "Retry" text will reload the video on Android. All perfect.

However on iOS, this exact same setup does not produce the same behaviour. With this setup, my `onTap` method is never called, `onLongPress` works ok but double tapping plays/pauses the video. If I wrap `BetterPlayer` in an `IgnorePointer` widget, the behaviour is the same as an Android and works as expected BUT the "Retry" text (if the video fails to load) is not clickable any more.

Question: how can I get the same behaviour on iOS than I have an Android?

**Comment by jhomlala** [comment]
Seems like some issue on Flutter iOS. This is something that I can't fix on Better Player. Anyway I've did some research and you may use RawGestureDetector to detect onTap over others GestureDetectors. 

```dart
import 'package:better_player/better_player.dart';
import 'package:flutter/gestures.dart';
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

class NormalPlayerPage extends StatefulWidget {
  @override
  _NormalPlayerPageState createState() => _NormalPlayerPageState();
}

class _NormalPlayerPageState extends State<NormalPlayerPage> {
  BetterPlayerController _betterPlayerController;

  @override
  void initState() {
    BetterPlayerConfiguration betterPlayerConfiguration =
        BetterPlayerConfiguration(
      aspectRatio: 16 / 9,
      fit: BoxFit.contain,
    );
    BetterPlayerDataSource dataSource = BetterPlayerDataSource(
      BetterPlayerDataSourceType.network,
      "http://cdn.theoplayer.com/video/elephants-dream/playlist.m3u8",
    );
    _betterPlayerController = BetterPlayerController(BetterPlayerConfiguration(
      autoDetectFullscreenDeviceOrientation: true,
      fit: BoxFit.cover,
      autoPlay: true,
      looping: true,
      controlsConfiguration: BetterPlayerControlsConfiguration(
          enableOverflowMenu: false, enableSkips: false),
    ));
    _betterPlayerController.setupDataSource(dataSource);
    _betterPlayerController.setControlsEnabled(false);
    //debugPaintSizeEnabled=true;
    super.initState();
  }

  bool showDetails = false;

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Normal player"),
      ),
      body: Stack(
        children: [
          RawGestureDetector(
            gestures: {
              AllowMultipleGestureRecognizer:
                  GestureRecognizerFactoryWithHandlers<
                      AllowMultipleGestureRecognizer>(
                () => AllowMultipleGestureRecognizer(),
                (AllowMultipleGestureRecognizer instance) {
                  instance.onTap = () => print('ON TAP ');
                },
              )
            },
            child: GestureDetector(
              onLongPress: (){
                print("LONG PRESS");
              },
              child: BetterPlayer(controller: _betterPlayerController),
            ),
          ),
        ],
      ),
    );
  }
}

class AllowMultipleGestureRecognizer extends TapGestureRecognizer {
  @override
  void rejectGesture(int pointer) {
    acceptGesture(pointer);
  }
}

```

Other option (which in my opinion is better than using multiple gesture detectors) is custom theme. Just copy what you need from cupertino or material controls and you should get rid of these strange issues with clicks.

**Comment by themadmrj** [comment]
I tried building a custom theme and copying most of the code from `better_player_cupertino_controls.dart`. In the end, when I removed the `GestureDetector`, the problem dissapeared but of course the Widget was not clickable anymore in fullscreen either...
However, simply setting the config to `playerTheme: BetterPlayerTheme.material` replicates the Android behaviour on iOS, which is what I wanted. To fix the wrong icons behing displayed, I just specified which icons to use depending on the Platform. Thanks for pointing out the theme idea, eventhough simply switching out the theme suggests that somewhere in `better_player_cupertino_controls.dart` something creates a different behaviour than material...
